### DISCUSSION TITLE
Browser wallet: Suggest UX interface improvements

### DISCUSSION THREAD

**Discussion by geopor** [tearust/teaproject#46] [Ideas]
_Add requests here for anything you think can be improved with the TEA browser wallet interface._

## Add the option to pay off interest and renew Genesis seed loans

Right now it's tedious to pay off entire loan, then go back to assets and take out a loan against same CML seed. Could have an option next to the **Pay off** icon to renew the loan by paying off interest due and adding to the loan term.

<img width="1176" alt="Screen Shot 2021-08-20 at 9 11 57 AM" src="https://user-images.githubusercontent.com/86096370/130475814-ac8581f8-2549-4aa1-aab4-71828795c1fb.png">

**Comment by geopor** [comment]
## Include a "Type" column in the auctions list

Right now you have to click through to find out what kind of seed it is. 

<img width="1110" alt="Screen Shot 2021-08-23 at 8 30 05 AM" src="https://user-images.githubusercontent.com/86096370/130476199-ae42bae4-42b8-43cd-b03d-4990613d2486.png">

**Comment by kevingzhang** [comment]
> _Add requests here for anything you think can be improved with the TEA browser wallet interface._
> 
> ## Add the option to pay off interest and renew Genesis seed loans
> Right now it's tedious to pay off entire loan, then go back to assets and take out a loan against same CML seed. Could have an option next to the **Pay off** icon to renew the loan by paying off interest due and adding to the loan term.
> 
> <img alt="Screen Shot 2021-08-20 at 9 11 57 AM" width="1176" src="https://user-images.githubusercontent.com/86096370/130475814-ac8581f8-2549-4aa1-aab4-71828795c1fb.png">

There will be a new Genesis Loan feature in next epoch. This feature will be included.
Other features would be: 
- compound interest, recalculated very 100 blocks
- variable interest rate calculated based on AMM curve, similar to Uniswap x*y=k

We will only take CML seeds as collateral in the next epoch, but in the future, we may take TApp tokens as well.

**Comment by geopor** [comment]
The following error was because of a bug:

<img width="515" alt="1" src="https://user-images.githubusercontent.com/86096370/130467919-d84e55b6-a992-4c0a-a750-22a83e4eb9a6.png">

But regardless, users should be able to retract bids. This is just one scenario - another is that they didn't realize they needed TEA for something else like paying off a loan.

**Suggestion**: Just as sellers get a 1 TEA penalty for retracting an auction, bidders should get a 1 TEA penalty for retracting a bid. But they should be allowed to do it, otherwise people will feel trapped by submitting a bid.

**Reply by kevingzhang** [reply]
Please create new topic for this error, otherwise this topic would be too long and no way to track.
the problem is not bidder cannot retract bids, it is because the CML186 is staking, so the deal did not go through. We have the issue and devs will take a look.

Regaring allow/disallow highest bidder to retract. I think we should not allow. The reason is that if there is only one T penalty, I can just put a highest bid that no one will beat, but I retract at the last moment so that a much lower bid (very likely from my buddy or myself)will win this bid. Let me give you an example. A CML worthy 2000 T, but the starting price is just 500. I put the bid at 501. then I put another bid at 10000000. Of course, no one could beat my super high price. but at the last moment, I retract the  1000000 bid with only one T penalty, so the original 501 bid win. So I spend only 502 T can get a CML worthy2000. This is the problem we want to prevent.

**Reply by geopor** [reply]
The error was just an example of an external issue causing bidders to want to retract bid, this issue is not for that.

In the above scenario, we could counter that type of scenario using the formula **your retracted bid amount** - **next highest bidder** = penalty to be paid. If there is no next highest bidder, then can subtract the auction's opening bid price from retracted bid amount to get penalty.

**Comment by kevingzhang** [comment]
I think the formula makes sense. and it is doable.
Where do you think the penalty should be sent to? to the next highest bidder or the current owner? or just burn?

**Reply by geopor** [reply]
I would vote to burn the penalty amount.

**Reply by kevingzhang** [reply]
this issue is taken care by https://github.com/tearust/teaproject/issues/48
